devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,371 @@
+	refresh(endpoint)
+}
+
+def refresh(endpoint) {
+	log.info "refresh called: endpint '${endpoint}' "
+	if(endpoint) {
+		secureEncap(zwave.basicV1.basicGet(), endpoint)
+	} else {
+		def commands = []
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1)
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 5)
+		sendCommands(commands,1000)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd){
+	log.info "ConfigurationReport called: ${cmd}"
+	def result = []
+	switch (cmd.parameterNumber) {
+		case 1:
+			state.tempOffset = cmd.scaledConfigurationValue
+			break
+		case 2:
+			state.humidityOffset = cmd.scaledConfigurationValue
+			break
+	}
+	result <<  createEvent(name: "configReport", value: "${state.tempOffset}, ${state.humidityOffset}%")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelEndPointReport cmd, endpoint = null) {
+	log.info "MultiChannelEndPointReport called: cmd '${cmd}'"
+	if(!childDevices) {
+		def type = isDawonWallSmartSwitch()
+		log.debug "MultiChannelEndPointReport: type '${type}'"
+		if (type) {
+			addChildSwitches(type)
+		} else {
+			log.debug "child endpoint=$cmd.endPoints"
+			addChildSwitches(cmd.endPoints)
+		}
+	}
+}
+
+private sendCommands(cmds, delay=1000) {
+	log.info "sendCommands called: cmds '${cmds}', delay '${delay}'"
+	sendHubCommand(cmds, delay)
+}
+
+private commands(commands, delay=200) {
+	log.info "commands called: commands '${commands}', delay '${delay}'"
+	delayBetween(commands.collect{ command(it) }, delay)
+}
+
+
+private secure(cmd) {
+	log.info "secure called"
+	if (zwaveInfo?.zw?.contains("s")) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else if (zwaveInfo?.cc?.contains("56")){
+		zwave.crc16EncapV1.crc16Encap().encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	log.info "encap called"
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+private secureEncap(cmd, endpoint = null) {
+	log.info "secureEncap called"
+	secure(encap(cmd, endpoint))
+}
+
+private changeSwitch(endpoint, value) {
+	log.info "changeSwitch called: vlaue: '${value}', endpoint: '${endpoint}'"
+	def result = []
+	if(endpoint) {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+		log.debug "changeSwitch: endpoint '${endpoint}', value: '${value}')"
+		result << child.sendEvent(name: "switch", value: value)
+	}
+	result
+}
+
+def childOnOff(deviceNetworkId, value) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(value, switchId)
+}
+
+def childRefresh(deviceNetworkId) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand refresh(switchId)
+}
+
+private isDawonWallSmartSwitch() {
```

Comment:
since the purpose of this function has changed, its name should also change. Perhaps `getNumberOfChildrenFromModel` or something like that.